src/App.tsx

```diff
@@ -1,24 +1,220 @@
+import { Footer } from './Components/Footer';
+import { TodoNotification } from './Components/TodoNotification';
+import { Header } from './Components/Header';
 
-const USER_ID = 0;
+const USER_ID = 10898;
+
+export const App: FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [filterOption, setFilterOption] = useState(FilterOption.All);
+  const [error, setError] = useState<string | null>(null);
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+  const [loadingItems, setLoadingItems] = useState<number[]>([]);
+  const [isLoadingAddTodo, setIsLoadingAddTodo] = useState(false);
+
+  const isAllCompleted = todos.every(todo => todo.completed);
+
+  const visibleTodos = useMemo(() => {
+    return todos.filter((todo) => {
+      switch (filterOption) {
+        case FilterOption.Active:
+          return !todo.completed;
+
+        case FilterOption.Completed:
+          return todo.completed;
+
+        default:
+          return FilterOption.All;
+      }
+    });
+  }, [filterOption, todos]);
+
+  useEffect(() => {
+    getTodos(USER_ID)
+      .then((todosFromServer) => {
+        setError(null);
+        setTodos(todosFromServer);
+      })
+      .catch(() => {
+        setError('Unable to upload todos');
+      });
+  }, []);
+
+  useEffect(() => {
+    let errorTimer: number;
+
+    if (error) {
+      errorTimer = window.setTimeout(() => {
+        setError(null);
+      }, 3000);
+    }
+
+    return () => {
+      clearTimeout(errorTimer);
+    };
+  }, [error]);
+
+  const addTodo = useCallback(async (title: string) => {
+    setIsLoadingAddTodo(true);
+
+    const newTodo = {
+      id: 0,
+      completed: false,
+      title,
+      userId: USER_ID,
+    };
+
+    setTempTodo({ ...newTodo });
+
+    try {
+      const createdTodo = await createTodo(newTodo);
+
+      setTodos(prevTodos => [...prevTodos, createdTodo]);
+    } catch (err) {
+      setError('Unable to add a todo');
+    } finally {
+      setTempTodo(null);
+      setIsLoadingAddTodo(false);
+    }
+  }, []);
+
+  const removeTodo = useCallback((todoId: number) => {
+    setLoadingItems(prevItems => [...prevItems, todoId]);
+
+    deleteTodo(todoId)
+      .then(() => {
+        setTodos(prevTodos => prevTodos.filter(
+          todo => todo.id !== todoId,
+        ));
+      })
+      .catch(() => {
+        setError('Unable to delete a todo');
+      })
+      .finally(() => {
+        setLoadingItems(prevItems => prevItems.filter(
+          id => id !== todoId,
+        ));
+      });
+  }, []);
+
+  const removeCompletedTodos = useCallback(() => {
+    const completedTodos = todos.filter(todo => todo.completed);
+
+    completedTodos.forEach(todo => removeTodo(todo.id));
+  }, [removeTodo, todos]);
+
+  const editTodo = useCallback(async (
+    todoId: number,
+    args: UpdateTodoArgs,
+  ) => {
+    setLoadingItems(prevItems => [...prevItems, todoId]);
+
+    const updatedTodo: UpdateTodoArgs = { ...args };
```

Comment:
not sure if you shold set type here, since it should inherit it from args variable

Reply:
yeap, you are right it my mis